httpserver/handler.go

```diff
@@ -4,10 +4,23 @@ import (
 	"net/http"
 	"time"
 
+	"github.com/flashbots/go-template/metrics"
 	"github.com/flashbots/go-utils/logutils"
 )
 
 func (s *Server) handleAPI(w http.ResponseWriter, r *http.Request) {
+	m := metrics.Float64Histogram(
```

Comment:
a line of documentation would be nice. this gets instantiated on every request and should not get reused?

Reply:
re. point 2, see the doc-comment at https://github.com/flashbots/go-template/pull/17/files#diff-d715f391f2ffa038c3167bfea1adf741c1f5e39f08349fbbacab780a594146a4R126-R128

```go
// Once created it's cached and reused further on. All subsequent calls
// to this method that use the same name will retrieve already created
// histogram from the cache.
```

so, no, the histogram is not instantiated on each request.  quite the opposite, it's re-used (based on its name), because that the whole point of it, it's an aggregate that keeps absorbing all the underlying statistics (total count + counts per bucket) until the scrape comes in, reads them all, and resets the statistics for until the next scrape.